proto/protovalidate/buf/validate/validate.proto

```diff
@@ -55,6 +53,94 @@ extend google.protobuf.FieldOptions {
   // Rules specify the validations to be performed on this field. By default,
   // no validation is performed against a field.
   optional FieldConstraints field = 1159;
+
+  // Specifies shared rules. When extending a standard constraint message, this
+  // adds additional CEL expressions that apply when the extension is used.
+  //
+  // ```proto
+  // extend buf.validate.Int32Rules {
+  //   bool is_zero [(buf.validate.shared_field).cel = {
+  //     id: "int32.is_zero",
+  //     message: "value must be zero",
+  //     expression: "rule ? this == 0 : ''",
+  //   }];
+  // }
+  //
+  // message Foo {
+  //   int32 reserved = 1 [(buf.validate.field).int32.(is_zero) = true];
+  // }
+  // ```
+  optional SharedFieldConstraints shared_field = 1160;
+}
+
+// `Constraint` represents a validation rule written in the Common Expression
+// Language (CEL) syntax. Each Constraint includes a unique identifier, an
+// optional error message, and the CEL expression to evaluate. For more
+// information on CEL, [see our documentation](https://github.com/bufbuild/protovalidate/blob/main/docs/cel.md).
+//
+// ```proto
+// message Foo {
+//   option (buf.validate.message).cel = {
+//     id: "foo.bar"
+//     message: "bar must be greater than 0"
+//     expression: "this.bar > 0"
+//   };
+//   int32 bar = 1;
+// }
+// ```
+message Constraint {
+  // `id` is a string that serves as a machine-readable name for this Constraint.
+  // It should be unique within its scope, which could be either a message or a field.
+  optional string id = 1;
+
+  // `message` is an optional field that provides a human-readable error message
+  // for this Constraint when the CEL expression evaluates to false. If a
+  // non-empty message is provided, any strings resulting from the CEL
+  // expression evaluation are ignored.
+  optional string message = 2;
+
+  // `expression` is the actual CEL expression that will be evaluated for
+  // validation. This string must resolve to either a boolean or a string
+  // value. If the expression evaluates to false or a non-empty string, the
+  // validation is considered failed, and the message is rejected.
+  optional string expression = 3;
+}
+
+// `Violations` is a collection of `Violation` messages. This message type is returned by
+// protovalidate when a proto message fails to meet the requirements set by the `Constraint` validation rules.
+// Each individual violation is represented by a `Violation` message.
+message Violations {
```

Comment:
I wouldn't expect these to be intermixed in order with the `*Constraints` definitions. These should likely be at the bottom of the file

Reply:
Moved `Violations` and `Violation` to end of file.